compiler/qsc_frontend/src/compile/preprocess.rs

```diff
@@ -131,7 +131,11 @@ fn matches_config(attrs: &[Box<Attr>], capabilities: TargetCapabilityFlags) -> b
         return true;
     }
     let mut found_capabilities = TargetCapabilityFlags::empty();
+    let mut disallowed_capabilities = TargetCapabilityFlags::empty();
 
+    // When checking attributes, anything we don't recognize (invalid form or invalid capability) gets
+    // left in the compilation by returning true. This ensures that later compilation steps, specifically lowering
+    // from AST to HIR, can check the attributes and return errors as appropriate.
     for attr in attrs {
```

Comment:
We need to handle `not Base` and `not Unrestricted`.

Reply:
Good catch! I'll fix the handling there and get test cases added.